pkcs12.go

```diff
@@ -582,9 +605,16 @@ func EncodeTrustStore(rand io.Reader, certs []*x509.Certificate, password string
 	})
 
 	var certBags []safeBag
-	for _, cert := range certs {
+	for i, cert := range certs {
+
+		subject := cert.Subject.String()
+		if len(friendlyNames) > 0 {
+			if friendlyNames[i] != "" {
```

Comment:
Is it worth checking length here or above? I assume the two are meant to be the same but that might not be clear to callers.

Also slightly unsure if skipping on an empty FriendlyName is right vs. returning an error to force callers to populate. Thoughts?

Reply:
yeah, I wasn't sure whether to enforce or not. Was thinking whether the `friendlyName` string and `x509.Certificate` should be bundled together in a struct, as that would make the API clearer imo, but this would be backwards incompatible.

To get around that, yes, we should make it clear that the lengths should be the same and that there should be a friendlyName string defined per cert (my reasoning for empty string was to default to the cert Subject, instead we could make the caller ensure this is set to make it explicit).

Reply:
I think you could get away with just one of them? Maybe allow it to be the empty string but require lengths to match, then you can just do `make([]string, len(certs))` above?

Reply:
As discussed offline, decided to go with a new struct instead, which makes it clear that a `FriendlyName` string is tied to a particular cert.

Should we also return an error if a `FriendlyName` field isn't populated? I haven't actually tested whether an empty string even works (which I doubt due to Java usually requiring the alias to be present).